server/service.go

```diff
@@ -345,15 +351,15 @@ func (m *BoostService) handleGetHeader(w http.ResponseWriter, req *http.Request)
 			}
 
 			blockHash := responsePayload.Data.Message.Header.BlockHash.String()
-			valueEth := weiBigIntToEthBigFloat(responsePayload.Data.Message.Value.BigInt())
+			valueEth := weiBigIntToEthBigFloat(responsePayload.Data.Message.Value.ToBig())
```

Comment:
If for any reason the response contains a nil `Value` field, this would segfault.
This is not the case right now as `U256Str` defaults to a 32-bytes array filled with zeros.

```go

func pocGetHeaderNilMessageValueGoBoostUtils() {
	responsePayload := new(types.GetHeaderResponse)
	responsePayload.Data = &types.SignedBuilderBid{
		Message: &types.BuilderBid{},
	}

	fmt.Println(responsePayload.Data.Message.Value)
}

func pocGetHeaderNilMessageValueAttestantTypes() {
	responsePayload := new(spec.VersionedSignedBuilderBid)
	responsePayload.Data = &bellatrix.SignedBuilderBid{
		Message: &bellatrix.BuilderBid{},
	}

	fmt.Println(responsePayload.Data.Message.Value)
}

func main() {
	// Prints the following:
	// [0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0]
	// <nil>

	pocGetHeaderNilMessageValueGoBoostUtils()
	pocGetHeaderNilMessageValueAttestantTypes()
}
```

What I would suggest (I can create one here because the corresponding lines were not edited) is to add a `.Value != nil` in the if statement at line 349.

Reply:
Thanks for the link, I did not realized it was already verified 👍